oml/utils/misc.py

```diff
@@ -148,7 +149,13 @@ def compare_dicts_recursively(d1: Dict, d2: Dict) -> bool:  # type: ignore
                 v, d2[k]
             ), f"The dictionaries differs at key {k}.\nDict_1 value: {v}\nDict_2 value: {d2[k]}"
         else:
-            assert d2[k] == v, f"Key name: {k}\nDict_1 value: {v}\nDict_2 value: {d2[k]}"
+            if isinstance(d2[k], torch.Tensor) and isinstance(v, torch.Tensor):
+                is_equal = torch.all(torch.isclose(d2[k], v))
+            elif isinstance(d2[k], float) and isinstance(v, float):
+                is_equal = math.isclose(d2[k], v, rel_tol=1e-6)
```

Comment:
Let's ensure consistency in tolerance values for numerical comparisons. Currently, `torch.isclose` defaults to `rtol=1e-05` and `atol=1e-08`, while `math.isclose` uses `rel_tol=1e-09` and `abs_tol=0.0`.  I suggest updating both functions to use the same values, specifically `rtol=1e-06` and `atol=1e-08`.

Reply:
agree